espnet/nets/pytorch_backend/transducer/transformer_decoder.py

```diff
@@ -295,3 +290,242 @@ def recognize_beam(self, h, recog_args, rnnlm=None):
             ]
 
         return nbest_hyps
+
+    def recognize_beam_nsc(self, h, recog_args, rnnlm=None):
```

Comment:
Is there some condition that both beam search methods produce identical results (maybe beam=1?). This would be a good unit test.

Reply:
Good question, I'll have to investigate! However it should be noted that: 1) NSC has a fixed number of expansions for all beam hypothesis and has local pruning after each expansion, 2) prefix search was removed in default beam search